sky/backends/cloud_vm_ray_backend.py

```diff
@@ -2803,6 +2854,49 @@ def _provision(
             if dryrun:
                 record = global_user_state.get_cluster_from_name(cluster_name)
                 return record['handle'] if record is not None else None
+
+            # We pass handle in the config dict for the new provisioner
+            if 'provision_metadata' in config_dict:
+                # New provisioner is used here.
+                handle = config_dict['handle']
+                provision_metadata = config_dict['provision_metadata']
+                resources_vars = config_dict['resources_vars']
+
+                # setup SkyPilot runtime after the cluster is provisioned
```

Comment:
The detailed comments should be included in the `provision_utils.post_provision_runtime_setup()`'s docstr as well.
```suggestion
                # Setup SkyPilot runtime after the cluster is provisioned
                # 1. Wait for SSH to be ready
                # 2. Mount the cloud credentials, skypilot wheel, and other necessary files to the VM
                # 3. Run setup commands to install dependencies
                # 4. Starting ray cluster and skylet.
```